packages/agent/src/main.ts

```diff
@@ -1,25 +1,24 @@
-import { ContentType, MedplumClient, normalizeErrorString } from '@medplum/core';
+import { Hl7Message, MedplumClient } from '@medplum/core';
 import { Bot } from '@medplum/fhirtypes';
-import { Hl7MessageEvent, Hl7Server } from '@medplum/hl7';
+import { Hl7Connection, Hl7MessageEvent, Hl7Server } from '@medplum/hl7';
 import { EventLogger } from 'node-windows';
-
-const log = new EventLogger({
-  source: 'MedplumService',
-  eventLog: 'SYSTEM',
-});
+import WebSocket from 'ws';
 
 export class App {
   readonly log: EventLogger;
   readonly server: Hl7Server;
+  readonly connections: Connection[] = [];
 
   constructor(readonly medplum: MedplumClient, readonly bot: Bot) {
     this.log = new EventLogger({
       source: 'MedplumService',
       eventLog: 'SYSTEM',
     });
 
-    this.server = new Hl7Server();
-    this.server.addEventListener('message', (event) => this.handler(event));
+    this.server = new Hl7Server((connection) => {
```

Comment:
I changed this from `addEventListener` to a callback, because the `Hl7Server` would receive messages before the event listener attached.  That meant the first message would often get dropped.  By using a callback, it becomes blocking, and the first message is not missed.

Alternatively, we could add message buffering to `Hl7Server`, but that has a bunch of tricky design trade offs (what does it mean to have multiple connection handlers?)